# Задача № 1

 Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
 
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy

Полученные значения должны быть равны.

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,

а затем с использованием функций из библиотек numpy и pandas.

In [46]:
import pandas as pd
import numpy as np

In [7]:
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

In [9]:
df = pd.DataFrame(list(zip(zp, ks)),

               columns =['zp', 'ks'])

In [10]:
df[['zp', 'ks']].corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

In [11]:
X = df['zp']
Y = df['ks']

MX = X.mean()
MY = Y.mean()

cov = ((X - MX) * (Y - MY)).sum() / (X.shape[0] - 1)
cov

10175.37777777778

In [14]:
np.cov(X, Y, ddof=1)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [15]:
corr = cov / (X.std() * Y.std())
corr

0.8874900920739164

In [16]:
df[['zp', 'ks']].corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

Ответ: ковариация: 10175.37777777778 ; коэффициент корреляции Пирсона: 0.8874900920739164

# Задача № 2

Измерены значения IQ выборки студентов,

обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [36]:
from scipy import stats

In [23]:
samples = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [24]:
n = samples.shape[0]
mean = samples.mean()
std = samples.std(ddof=1)

n, mean, std

(10, 118.1, 10.54566788359614)

так как дисперсия неизвестна, то ищем квантили через t-распределение Стьюдента

In [25]:
p = 0.95
alpha = 1 - p

t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)

t1, t2

(-2.2621571627409915, 2.2621571627409915)

In [30]:
(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))

(110.55608365158724, 125.64391634841274)

# Задача № 3

Известно, что рост футболистов в сборной распределен нормально

с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,

среднее выборочное составляет 174.2. Найдите доверительный интервал для математического

ожидания с надежностью 0.95.

In [43]:
n = 27
std = 5
mean = 174.2

так как распределение нормальное, то квантили можно найти так:

In [44]:
p = 0.95
alpha = 1 - p

t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)

t1, t2

(-1.959963984540054, 1.959963984540054)

отсюда доверительный интервал находим по формуле:

In [45]:
(mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))

(172.3140237765397, 176.08597622346028)